### PR TITLE
fix issue #4: Sorting fails for files with only one timestep

### PR DESCRIPTION
**Pull request by jk-rieck**
Fixing issue #4 : Before squeezing, check for dimensions of size 1 (which would be squeezed) and exclude dimension "t" from these, when it exists. Then squeeze the remaining dimensions of size 1.

### CODE REVIEW

File: xorca/lib.py [hunk 1]
```diff
@@ -58,7 +58,17 @@ def trim_and_squeeze(ds,
...
+    to_squeeze = []
+    for dimensions in ds.dims:
+        if ds[dimensions].size == 1:
+            to_squeeze.append(dimensions)
+    
+    if "t" in to_squeeze:    
+        t_index = to_squeeze.index("t")
+        del to_squeeze[t_index]
+        
+    ds = ds.squeeze(dim = to_squeeze)
```
Comment (willirath) [inline xorca/lib.py]: We should to this with a list comprehension:

```python
to_squeeze = [dim for dim in ds.dims if ((ds[dim].size == 1) and (dim is not "t")]
```

And we should make sure that this is tested.  Let's include a singleton dimension "t" here https://github.com/willirath/xorca/blob/0447e26428ed5f75ae5711b697d0562d7c6f5646/xorca/tests/test_lib.py#L24 and `assert` that it's still present in the resulting data set.

**Comment by willirath** [comment]
Hhhmmm. I am re-running the tests.  Not sure why theire failing.

**Comment by jk-rieck** [comment]
I narrowed down the problem: 
There is a missmatch between the dimensions we assign to the mesh_mask variables in test_mesh_mask.py and the dimensions of the mesh_mask variables in orca_names.py. In test_mesh_mask.py they all have dimension "t" (as they do in the original files), in orca_names.py the dimension "t" is missing.
The tests are successful with 189de0355bc503b76a405ba1bf862975c4d8d8ee
However, we have a new problem (I guess) because then, all `e3t` etc. will have a time dimension and lib.set_time_independent_vars_to_coords() will not be effective.... Maybe you can have a look, @willirath , you are way more familiar with the code...

**Comment by willirath** [comment]
We're seeing a combination of issues here: Some are rooted in the general design of `xorca` and some of which are due to problems in the tests themselves.  (All are my fault :) )

1. We can't handle time dependent metric fields at the moment.  This is mainly because the auxiliary files have a degenerate time axis that does not refer to a real point in time.  So we (need to?) squeeze these out for now. (The metric fields can, in principle, be time dependent: With `vvl`, we can have a time varying vertical grid.)

2. For some the tests, I've used auxiliary files like data files.  This is fine from a coverage point of view, beceause processing the aux files as real data files also passes the whole pipeline from the raw data sets to a final integrated data set.  It is, however, not very representative of a real use case where, e.g., the time axes of the real data files should be treated differently from those of the aux files.

For solving this (in a sustainable way), we could do the following distinguish meaningful time axes from the broken ones we find in aux files and only squeeze out the broken ones.

We could add another condition to the list comprehension building the list of dims to squeeze out that checks if there is a variable `ds.data_vars[dim]` that is a valid time axis.

**Comment by jk-rieck** [comment]
In commit d7dc10547a77e8edf7628a2cada3648d34ac1580 , I tried to incorporate your suggestion and now check whether the dimension `t` is of `dtype` `datetime64`. This should also allow aux_files to have a time axis in the future (if it is valid). At least locally, the tests were successful!

**Comment by willirath** [comment]
Looks good!

I think, we'll run into trouble only checking for np.datetime64 at some point, because xarray falls back to other datetime objects in some cases. But let's deal with this in a different PR.

Thanks!

**Pull request merged**